### ISSUE TITLE
hibernate-vector integration (through version management OR extension)

### ISSUE THREAD

**Issue by lordofthejars** [quarkusio/quarkus#41453]
### Description

Currently (or at least what I've found), to use the `hibernate-vector` dependency, you need to add the version of Hibernate manually. 

```
<dependency>
            <groupId>org.hibernate.orm</groupId>
            <artifactId>hibernate-vector</artifactId>
            <version>6.4.7.Final</version> <!-- Sets to Hibernate version used by Quarkus -->
        </dependency>
```

This means you need to explore which Hibernate version is transitively imported by Panache and then hardcoded in the pom.
So, if I update the Quarkus version to a newer version, and the Hibernate version used by Panache is updated too, then my pom might have a divergence in versions.

### Implementation ideas

Create an extension that just imports the correct `hibernate-vector` or a Panache bom so the version of Hibernate is implicitly set.

**Comment by gsmet** [comment]
If it works out of the box, we can just include it in the BOM. Does it though? :)

**Comment by lordofthejars** [comment]
Yes it works, the only thing is that you need to know beforehand the Hibernate version used by Panache.

**Comment by yrodiere** [comment]
Right, we don't necessarily need a separate extension, a BOM entry would be fine as a first step.

The patch is trivial, testing... not so much. I'd need a separate integration test module just for this... including a native build. @gsmet are you fine with that?

**Comment by yrodiere** [comment]
> The patch is trivial, testing... not so much. I'd need a separate integration test module just for this... including a native build. [@gsmet](https://github.com/gsmet) are you fine with that?

Let's assume @gsmet is fine with that and add an IT module for pgvector, at least :)

**Comment by MichielDx** [comment]
Is there an update on this?
